resources/metadata/npmExecuteScripts.yaml

```diff
@@ -26,6 +26,16 @@ spec:
       - name: source
         type: stash
     params:
+      - name: tool
+        type: string
+        description: |
+          The to run the script. Can be one of: auto, npm, yarn, pnpm.
+          Default is 'auto' to auto-detect the tool based on the lock file.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: "auto"
```

Comment:
should we perhaps default to `npm`  and simplify the logic for auto . if i understand auto is just a placeholder that will get replaced with `npm / yarn / pnpm ... ` . it maybe a little less confusing for end users as well ?

Reply:
There is no need to worry about regressions with `auto`. 
If users have a `package-lock.json` in their repo it will select `npm` if they don't have any `lock` file it  will fall-back to `npm` behaviour.